base/deprecated.jl

```diff
@@ -1470,4 +1470,7 @@ end
 @deprecate (|)(A::AbstractArray, b::Number)         A .| b
 @deprecate (|)(A::AbstractArray, B::AbstractArray)  A .| B
 
+# Calling promote_op is likely a bad idea, so deprecate to its convenience wrapper promote_eltype_op
```

Comment:
Errr... should read "so deprecate its convenience wrapper". Does that make sense?